ca_tools/scripts/action_server.py

```diff
@@ -0,0 +1,92 @@
+#! /usr/bin/env python
+
+import rospy
+import math
+import actionlib
+from geometry_msgs.msg import Point
+from nav_msgs.msg import Odometry
+from std_msgs.msg import Bool
+from ca_tools.msg import FollowLineAction, FollowLineFeedback, FollowLineResult
+
+
+class RobotAction(object):
+    _ERROR_TOLERANCE = 0.1
+
+    def __init__(self, name):
+        self._curr_dist = 0.0
+        self._current_pos = Point()
+        self._last_pos = Point()
+        self._gts_first = True
+        self._feedback = FollowLineFeedback()
+        self._result = FollowLineResult()
+        self._gts_subscriber = rospy.Subscriber("/create1/gts", Odometry, self.gts_callback)
+        self._action_name = name
+        self._as = actionlib.SimpleActionServer(
+            self._action_name, FollowLineAction, execute_cb=self.execute_cb, auto_start = False)
+        self._as.start()
+        self._goal_reached = False
+        self._curr_time = rospy.Time.now()
+        self._r = rospy.Rate(2)
+
+
+    def gts_callback(self, data):
+        if self._gts_first:
+            self._current_pos.x = data.pose.pose.position.x
+            self._current_pos.y = data.pose.pose.position.y
+            self._gts_first = False
+        else:
+            self._last_pos.x = self._current_pos.x
+            self._last_pos.y = self._current_pos.y
+            self._current_pos.x = data.pose.pose.position.x
+            self._current_pos.y = data.pose.pose.position.y
+            self._curr_dist += math.hypot(abs( self._last_pos.x - self._current_pos.x), self._last_pos.y -
+                                     self._current_pos.y)
+
+    def goal_reach_control(self,_goal_point):
+        _x_limit = (self._current_pos.x <= (_goal_point.x + self._ERROR_TOLERANCE)) and (self._current_pos.x
+        >= (_goal_point.x - self._ERROR_TOLERANCE))
```

Comment:
My approach was to give the goal some tolerance, so it increments the area that is considered the "goal point", and since checking the the position can be done by just checking the /gts, that seems to me an easier but still reliable solution.